Reinstate Google Form approach, update form as necessary

fureigh opened the pull request:
Work in progress, not yet ready for review.

Remaining steps:
* [x] @hestherchang will update the form content. :tada:
* [x] @fureigh will restore the application confirmation page (`apply-thank-you.html`).
* [x] @hestherchang will review the application confirmation page and update content as appropriate.
* [x] @fureigh will check for and add any other necessary elements from that setup.
* [x] @fureigh will check whether we need anything from later than 9beed21.
* [ ] ~@fureigh will push to staging; both will verify.~ (see #339)

Closes https://github.com/18F/joining-18f/issues/334.

fureigh committed:
Restore old Google Form application

From 9beed21. Next step is for @hestherchang to update it. :tada:

https://github.com/18F/joining-18f/issues/334

fureigh commented on pages/apply.html:
```diff
@@ -0,0 +1,414 @@
+        window.location.href = './apply-thank-you.html';
+      })}"></iframe>
+
+    <form action="https://docs.google.com/a/gsa.gov/forms/d/1OCtPVLMP8O5nfDnGTJZngxFxZR4EvFcYSqJ-thHxojc/formResponse" method="post"
```
@hestherchang Replace the contents of the `action` field with the URL of the new application.

fureigh commented on pages/apply.html:
```diff
@@ -0,0 +1,414 @@
+        <p>Any individual who is not currently a GSA employee or contractor and who is a U.S. citizen, a non-citizen national of the U.S. or holds a valid green card is eligible to apply.</p>
+
+        <div class="usa-input">
+          <input type="checkbox" name="entry.1377828897" value="Verify that you meet the above eligibility requirements" id="group_1377828897_1" role="checkbox" class="ss-q-checkbox">
```
@hestherchang Here you'll want to update the contents of `name` and `id`.

fureigh commented on pages/apply.html:
```diff
@@ -0,0 +1,414 @@
+
+        <div class="usa-input">
+          <input type="checkbox" name="entry.1377828897" value="Verify that you meet the above eligibility requirements" id="group_1377828897_1" role="checkbox" class="ss-q-checkbox">
+          <label for="group_1377828897_1">Verify that you meet the above eligibility requirements</label>
```
@hestherchang ...and here it's the `for` field.

fureigh commented on pages/apply.html:
```diff
@@ -0,0 +1,414 @@
+        <h2><legend>Contact Information:</legend></h2>
+
+        <div class="usa-input name">
+          <label class="ss-q-item-label" for="entry_1741264393">Name <span class="required">*</span></label>
```
@hestherchang Basically anything that starts with `entry` or `group`. :)

fureigh replied:
...and make any content changes (if `Name` here was changed to `Full name`, for instance). I can check the `aria-label`s if you'd like; they need to match but also have those spaces.

fureigh committed:
Restore application confirmation page

From 9beed21.

https://github.com/18F/joining-18f/issues/334

fureigh commented:
@hestherchang I added the application confirmation page (`apply-thank-you.html`). When you have a chance, take a look and confirm that the timeline for response is still accurate. Thanks!

hestherchang committed:
updated fields to reflect new application form

hestherchang commented:
@fureigh I'm done updating the form content. Let me know what you think!

hestherchang commented:
@fureigh apply-thank-you.html looks good. thank you for doing that.

fureigh committed:
Update destination of form submissions

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Add stub page for product lead position

https://github.com/18F/joining-18f/issues/334

fureigh commented on _positions/product-lead.md:
```diff
@@ -0,0 +1,42 @@
+team: 18F
+active: true
+---
+**GS Grade Level:** 13
```
@hestherchang You can use this page for the product lead position listing. I updated the metadata (everything above the `---`), so you can start here! :)

hestherchang replied:
thank you! there are some lingering questions about the approved job posting so i'm waiting for some answers

fureigh replied:
Cool. I'll rename this file in the meantime.

hestherchang commented on pages/apply.html:
```diff
@@ -42,18 +42,31 @@
+          <p>
+          <label class="ss-q-item-label" for="entry_1741264393">First Middle Last Name <span class="required">*</span></label>
+          <br/>
+            If no middle name (NMN), write "NMN" between your first and last name. 
```
this is supposed to be a "helpful note" - not sure if it is "coded" correctly

fureigh replied:
The [last version of this form](https://docs.google.com/a/gsa.gov/forms/d/e/1FAIpQLScQgph3O99fB0Z9EYs8sKTtth3rX2-_slLFxS0lc_hu3D9AcQ/viewform) simply requested 'Name'. If there's not a demonstrated need to do otherwise here, I strongly recommend we leave it at 'Name', leaving any further elaboration for the SF-86.

hestherchang commented on pages/apply.html:
```diff
@@ -152,26 +165,27 @@
           <input type="checkbox" name="entry.1263390573" value="Verify your ability to work full-time" id="group_1263390573_1" role="checkbox" class="ss-q-checkbox">
-          <label for="group_1263390573_1">Verify your ability to work full-time</label>
+          <label for="group_1263390573_1">I am able to work full time (40 hours/week).</label>
+          <label for="group_1263390573_2">I am not able to work full time.</label>
```
not sure if this option will show up - i had to create my own "label for"...." line

fureigh replied:
Please change this from a checkbox to a radio button on the Google form. That way the applicant can select one and only one answer.

fureigh replied:
Separately, in case you're curious: the display issue was due to not encapsulating these in a list (`<ul>` and `<li>`).

hestherchang commented on pages/apply.html:
```diff
@@ -42,18 +42,31 @@
           <input type="checkbox" name="entry.1377828897" value="Verify that you meet the above eligibility requirements" id="group_1377828897_1" role="checkbox" class="ss-q-checkbox">
-          <label for="group_1377828897_1">Verify that you meet the above eligibility requirements</label>
+          <label for="group_1377828897_1">I, the applicant, verify that I meet the above eligibility requirements.</label>
+          <label for="group_1377828897_2">I, the applicant, verify that I do not meet the above eligibility requirements and will not be completing the rest of this application.</label>
```
not sure if this multiple choice option will show up

fureigh replied:
@hestherchang Please use the Google form admin interface to make this a radio button. (Rationale is explained below.) Thanks!

hestherchang commented on pages/apply.html:
```diff
@@ -42,18 +42,31 @@
+        </div>
+        
+        <div class="usa-input preferredname">
+          <label class="ss-q-item-label" for="entry_732113689">Preferred Name <span class="required">*</span></label>
```
i also added this and not sure if its "coded" correctly - and this is an optional question

fureigh committed:
Rename 'product lead' to 'product manager'

Per @hestherchang's feedback.

https://github.com/18F/joining-18f/issues/334

fureigh commented:
Thanks for the notes, @hestherchang! I'll finish taking a look.

fureigh commented on _positions/product-manager.md:
```diff
@@ -0,0 +1,42 @@
+team: 18F
+active: true
+---
+**GS Grade Level:** 13
```
@hestherchang This is the new spot to make changes.

That `pay-grades` part of the permalink is a holdover; I'll update the URL to `positions/product-manager/`.

fureigh committed:
Update product manager position's permalink

https://github.com/18F/joining-18f/issues/334

fureigh commented on pages/apply.html:
```diff
@@ -0,0 +1,364 @@
+<link rel="stylesheet" type="text/css" href="../assets/css/all.css">
+<link rel="stylesheet" type="text/css" href="../assets/css/form.css">
+
+<script type="text/javascript" src="../assets/js/jquery-1.11.3.min.js"></script>
```
The Standards are [now using version 2.2.0](https://github.com/18F/web-design-standards/blob/b953590cbd9712bd22b6477893b9de10957ba574/package.json). May update this accordingly, reviewing the markup for continued compatibility.

hestherchang replied:
ok 👍  thank you

fureigh commented on pages/apply.html:
```diff
@@ -42,18 +42,31 @@
+        
+        <div class="usa-input preferredname">
+          <label class="ss-q-item-label" for="entry_732113689">Preferred Name <span class="required">*</span></label>
+          <input type="text" name="entry.732113689" value="" class="ss-q-short" id="entry_732113689" dir="auto" aria-label="Preferred Name" aria-required="true"  data-parsley-trigger="change"  title="Preferred Name" required />
```
I still recommend leading with preferred name and considering leaving out 'legal name'. If the name setup stays this way, I recommend making this field optional so your results don't get cluttered up with people saying "N/A".

fureigh commented on pages/apply.html:
```diff
@@ -201,69 +215,22 @@
-          <label class="ss-q-item-label" for="entry_677498968">Please indicate your tertiary skillset.</label>
-          <select name="entry.677498968" id="entry_677498968" aria-label="Please indicate your tertiary skillset. Optional. Your response here will be used primarily in assigning reviewers for your application. ">
+          <select name="entry.227038756" id="entry_227038756" aria-label="What position and GS level are you applying to? GS level is the pay grade. Please see &quot;Government pay grades explained:&quot; https://pages.18f.gov/joining-18f/pay-grades/ " aria-required="true" required=>
             <option value=""></option>
```
@hestherchang Should this include a GS 13 option?

fureigh commented on pages/apply.html:
```diff
@@ -364,7 +314,7 @@
       <div class="usa-alert-body">
         <h1 class="usa-alert-heading">Thank you!</h1>
-        <p class="usa-alert-text">Your form submission has been successfully submitted.</p>
+        <p class="usa-alert-text">Thank you for your application. We'll be in touch if there is a fit. For questions, please email us at join18f@gsa.gov.</p>
```
@hestherchang I'll check whether the rest of this content is already covered on the confirmation screen.

In case it's not: applicants usually expect to hear back either way. Is it true that the Talent Team plans to reply only if they're interested in proceeding?

fureigh commented on pages/apply.html:
```diff
@@ -316,28 +283,11 @@
-
-          <label class="ss-q-item-label" for="entry_1108678713">Tell us why you&#39;re interested in this role.</label>
-          <textarea name="entry.1108678713" rows="8" cols="0" class="ss-q-long" id="entry_1108678713" dir="auto" aria-label="Tell us why you&#39;re interested in this role. "></textarea>
+          <label class="ss-q-item-label" for="entry_1108678713">Tell us why you&#39;re interested in the role you're applying for.</label>
```
Let's make both of these fields required.

fureigh committed:
Adjust fields' copy, generally to match the Google form

Also includes making 'Preferred name' an optional, not required,
field.

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Change some fields from checkboxes to radio buttons

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Add indicators for remaining required fields

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Copy edit: US --> U.S.

https://github.com/18F/joining-18f/issues/334

fureigh commented on _positions/product-manager.md:
```diff
@@ -0,0 +1,42 @@
+team: 18F
+active: true
+---
+**GS Grade Level:** 13
```
@hestherchang The form currently requires the applicant to select either `GS 14` or `GS 15`. This listing shows only `GS 13`. Please clarify the intended GS levels.

hestherchang replied:
intended level - gs14 and gs15

fureigh commented on pages/apply.html:
```diff
@@ -0,0 +1,380 @@
+        <div class="usa-input">
+          <ul class="usa-unstyled-list">
+            <li>
+              <input type="radio" name="entry.1377828897" value="Verify that you meet the above eligibility requirements" id="group_1377828897_1" role="radio" class="ss-q-radio" aria-required="true">
```
@hestherchang Please use the Google admin form to change this field from checkboxes to radio buttons. That way the applicant can select one but only one option. Thanks!

fureigh replied:
In case you're curious, the display issue was due to not encapsulating the options in `<ul>` (unordered list) and `<li>` (list item) tags.

hestherchang replied:
ooooo! good catch!! i changed it to multiple choice so only one answer can be selected

fureigh commented on pages/apply.html:
```diff
@@ -0,0 +1,380 @@
+        </div>
+
+        <div class="usa-input preferredname">
+          <label class="ss-q-item-label" for="entry_732113689">Preferred name</label>
```
I strongly recommend we just have one name field, synonymous with `Preferred name`, and leave the legal question for a later phase of the process. This recommendation is because as far as I'm aware, there isn't a need to have someone's legal name at this point, the legal name question risks introducing unnecessary confusion into future communications, and I promise you it risks making some applicants uncomfortable to such a degree that they won't apply.

If we do proceed with two fields, however, I recommend making this one optional so your results don't get cluttered up with a lot of `N/A`s or the equivalent.

hestherchang replied:
I thought I changed this to be similar to "Preferred Name". I want it to read "First Middle Last Name" and a note that says "if no middle name, please write "NMN"), with no mention of legal name.

fureigh commented on pages/apply.html:
```diff
@@ -0,0 +1,380 @@
+          <p>GS level is the pay grade. Please see &ldquo;<a href="https://pages.18f.gov/joining-18f/pay-grades/">Government pay grades explained</a>.&rdquo;</p>
+
+          <select name="entry.227038756" id="entry_227038756" aria-label="What position and GS level are you applying to? GS level is the pay grade. Please see &quot;Government pay grades explained:&quot; https://pages.18f.gov/joining-18f/pay-grades/ " aria-required="true" required=>
+            <option value=""></option>
```
Should GS-13 be included as well? (See also: my question on the position description.)

hestherchang replied:
Negative - we're only doing GS14 and GS15 for this role

fureigh commented on pages/apply.html:
```diff
@@ -0,0 +1,380 @@
+
+        <div class="usa-input">
+
+          <label class="ss-q-item-label" for="entry_227038756">What position and GS level are you applying to? <span class="required">*</span></label>
```
Copy edit suggestion: "To what position and GS level are you applying?"

hestherchang replied:
I'm cool with that

hestherchang replied:
I changed it in the google form

fureigh commented on pages/apply.html:
```diff
@@ -0,0 +1,380 @@
+
+        <div class="usa-input">
+
+          <label class="ss-q-item-label" for="entry_510169774">Why do you want to join 18F?<span class="required">*</span></label>
```
Let's make these two fields required too. (This is controlled via the Google form admin interface.)

hestherchang replied:
Another good catch! Thank you - I changed it on the form so both are mandatory.

fureigh commented on pages/apply.html:
```diff
@@ -0,0 +1,380 @@
+
+          <ul class="usa-unstyled-list">
+            <li>
+              <input type="radio" name="entry.1263390573" value="Verify your ability to work full-time." aria-label="Verify your ability to work full-time." id="group_1263390573_1" role="radio" class="ss-q-radio" aria-required="true" required>
```
Please change this to a radio button on the Google side.

hestherchang replied:
Another good catch!! 🎣  thank you!

fureigh reviewed (commented):
Hi @hestherchang,

I modified several pieces to better correspond to the Google form. I also noted a few places where I believe the Google form should be updated. Please review and reply as you're able.

I need to finish verifying that the form fields' IDs and such all correspond to their equivalents in the Google form, but I'm sending these content and configuration questions over in the meantime.

One more question: how do you want the applicant to find 1) the position description and 2) the application form?

fureigh commented on pages/apply.html:
```diff
@@ -0,0 +1,380 @@
+   <div class="usa-alert usa-alert-success">
+      <div class="usa-alert-body">
+        <h1 class="usa-alert-heading">Thank you!</h1>
+        <p class="usa-alert-text">Thank you for your application. We'll be in touch if there is a fit. For questions, please email us at join18f@gsa.gov.</p>
```
I'll check whether the existing confirmation page sufficiently covers this. In the meantime, @hestherchang: is it true that the Talent Team plans to reply only if they want to proceed with an applicant? I ask because most applicants expect to hear back either way.

fureigh replied:
(Pardon these questions displaying on the all-changes page instead of in response to your commit, @hestherchang. I got in an argument with GitHub's pull request review interface.)

hestherchang replied:
@fureigh  It is not true - Talent will respond either way. I'm not sure why I wrote it like that. How about replacing "if there is a fit" with "soon"? 

and no worries about the comments - I like this!

fureigh replied:
@hestherchang Glad to hear it all around! :)

Here's what we have on the form confirmation screen:

> Your form submission has been successfully submitted. We'll review your application shortly and get back to you within three to five weeks.

Is that timeframe still accurate?

hestherchang commented:
@fureigh  In response to... "One more question: how do you want the applicant to find 1) the position description and 2) the application form?" 

Let's talk about it on our call tomorrow! I want it very accessible but not sure what that looks like/what my options are, etc. Thank you soooo^infinity much for your help!

hestherchang committed:
Update product-manager.md

final job posting! would appreciate another look

hestherchang commented on _positions/product-manager.md:
```diff
@@ -0,0 +1,115 @@
+**Work Schedule:** Full Time<br>
+
+**Job Announcement Number:** 17001-Product Manager<br>
+**Open:** DATE<br>
```
we'll need to update this before we put it up (and the close date below - which is 5 calendar days inclusive of the open date)

fureigh replied:
Thanks for noting. I hate to suggest further delay, but if it's five calendar days inclusive of the open date, might we get more applicants if we waited until after the holidays to post this?

hestherchang committed:
Update product-manager.md

updated # of vacancies from 1 to 3

fureigh commented on _positions/product-manager.md:
```diff
@@ -0,0 +1,115 @@
+active: true
+---
+##Basic Information:
+**Title:** Product Manager  <br>
```
For future reference, Markdown files (files that have a `.md` extension, like this one) don't use `<br>`. All they need for a linebreak is a linebreak. I'll go ahead and take these out.

hestherchang replied:
👍

fureigh commented on _positions/product-manager.md:
```diff
@@ -0,0 +1,115 @@
+**Supervisory Status:** No<br>
+**Security Clearance:** Public Trust - background investigation required.<br>
+
+Do you have a passion for public service? The General Services Administration (GSA) is looking for you! GSA is a federal government agency whose mission is to bring the best value in real estate, acquisition, and technology services to the American people.<br>
```
@hestherchang I suggest adding one more sentence to connect the dots and explicitly note that 18F is part of GSA.

hestherchang replied:
Sure, we can add something at the end like "18F is an organization inside GSA's branch of technology (Technology Transformation Service (TTS))".

fureigh commented on _positions/product-manager.md:
```diff
@@ -0,0 +1,115 @@
+
+###Specialized Experience:
+**GS-14**
+To qualify at a GS-14 level you must demonstrate at least one year of specialized experience at the next lower grade level GS-13, or equivalent in the federal service:
```
Might be helpful to link to 'Government pay scales, explained', either on the 'GS-___'s or once separately. This information is available elsewhere on the site but we can't assume visitors to this page will have seen it.

hestherchang replied:
The info on that page is not the most relevant to this role because it refers to what the Innovation Specialist looks like at GS13-15, based off the Innovation Specialist PD. We can link to the "govt pay grades explained" as a reference but I'd want to note clearly that it is only for reference.

fureigh commented on _positions/product-manager.md:
```diff
@@ -0,0 +1,115 @@
+##Benefits
+GSA has been repeatedly named as one of the 'Best Place to Work' in the Federal Government. 
+
+GSA offers its employees a wide range of benefits including: Federal health insurance plans (choose from a wide range of plans); Life insurance coverage with several options to choose from; Leave policies to help you take care of your personal, recreational and health care needs; Thrift Savings Plan (similar to a 401(k) plan); Flexible work schedules and telework; Transit and child care subsidies ...
```
Mind if I make these bullet points?

fureigh committed:
Copy edit (position and GS level)

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Lowercase a bunch of capitalized words

...per 18F's content guide.

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Remove remaining `<br>`s from Markdown file

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Copy edits to 'number of vacancies'

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Add application link to end of position description

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Add Product Manager position to open roles page

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Update name fields per conversation

The problem was that people have been putting in just one word
(ex: 'Dave') despite using more than one word as their usual name,
and that's been making things confusing for the Talent Team.
Hopefully this will be clearer.

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Revert application confirmation message

...from the longer, friendlier one to a simpler one, because the
application confirmation screen should include the relevant text.

https://github.com/18F/joining-18f/issues/334

fureigh commented:
Hi @hestherchang,

I asked a few questions and pushed a set of changes. Unfortunately, [staging is down](https://github.com/18F/joining-18f/issues/339), so we'll have to just preview locally.

I also did a test form submission. Can you confirm whether that came through? (Please also grant access to the spreadsheet that collects Google form responses, when you have a moment, for future ease of troubleshooting. Thanks!)

hestherchang commented:
I did not receive the test form submission and access granted

hestherchang commented:
@fureigh  I see in my email you sent this message but I can't find it anywhere in Github... 
---
@hestherchang Glad to hear it all around! :)

Here's what we have on the form confirmation screen:

Your form submission has been successfully submitted. We'll review your application shortly and get back to you within three to five weeks.

Is that timeframe still accurate?
---
Yes, the time frame is still accurate.

hestherchang commented:
responding to another comment i see in my email but not here.. 

"+##Benefits
+GSA has been repeatedly named as one of the 'Best Place to Work' in the Federal Government. 
+
+GSA offers its employees a wide range of benefits including: Federal health insurance plans (choose from a wide range of plans); Life insurance coverage with several options to choose from; Leave policies to help you take care of your personal, recreational and health care needs; Thrift Savings Plan (similar to a 401(k) plan); Flexible work schedules and telework; Transit and child care subsidies; and Training and development
Mind if I make these bullet points?"

Sure, you can make these to bullets.

fureigh committed:
Add 18F-and-GSA explainer line

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Cleanup: remove Lever-specific assets

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Update apply link

This may be an artifact of testing locally.

https://github.com/18F/joining-18f/issues/334

fureigh committed:
Add application open and close dates

https://github.com/18F/joining-18f/issues/334
